assessmentModel/src/commonMain/kotlin/serialization/Serialization.kt

```diff
@@ -30,6 +32,38 @@ object Serialization {
     }
 }
 
+/**
+ * The [ModuleInfoProvider] provides methods to lookup the necessary [ResourceInfo] and [Json] to
+ * load an [Assessment] using the provide the platform specific [FileLoader].
+ */
+interface ModuleInfoProvider {
+
+    val fileLoader: FileLoader
+
+    fun getResourceInfo(assessmentIdentifier: String): ResourceInfo
+
+    fun getJsonDecoder(assessmentIdentifier: String?): Json
+}
+
+/**
+ * Default implementation of [ModuleInfoProvider] which can be extended to provide custom json decoders
+ * and [ResourceInfo].
+ */
+open class ModuleInfoProviderImpl(
+    override val fileLoader: FileLoader,
+    private val defaultResourceInfo: ResourceInfo
+): ModuleInfoProvider {
+
+    override fun getResourceInfo(assessmentIdentifier: String): ResourceInfo {
+        return defaultResourceInfo
+    }
+
+    override fun getJsonDecoder(assessmentIdentifier: String?): Json {
+        return Serialization.JsonCoder.default
+    }
+
+}
+
```

Comment:
I would suggest the following:

First, refactor `Assessment` to pull out the bits which come from the server.

```
interface AssessmentInfo {

    /**
     * The identifier for the assessment.
     */
    val identifier: String

    /**
     * The [versionString] may be a semantic version, timestamp, or sequential revision integer.
     */
    val versionString: String?

    /**
     * The [schemaIdentifier] is used to allow a group of [Assessment] models to all map to the same
     * table in a data base.
     */
    val schemaIdentifier: String?
}
```

Which makes the Assessment look like this:
```
interface Assessment : BranchNode, ContentNode, AssessmentInfo
``` 

Then change the `ModuleInfoProvider` to be:
```
/**
 * The [ModuleInfoProvider] provides methods to lookup the necessary [ResourceInfo] and [Json] to
 * load an [Assessment] using the provide the platform specific [FileLoader].
 */
interface ModuleInfoProvider {

    val fileLoader: FileLoader

    fun getRegisteredResourceInfo(assessmentInfo: AssessmentInfo): ResourceInfo? = null

    fun getRegisteredJsonDecoder(assessmentInfo: AssessmentInfo): Json? = null

    /**
     * Allows the provider to return an [Assessment] (for the given [assessmentInfo]) that is not
     * encoded as a JSON serialized object.
     */
    fun getRegisteredAssessment(assessmentInfo: AssessmentInfo): Assessment? = null
}
```

And finally:
```
/**
 * A [TransformableAssessment] is a placeholder that can be used to contain just enough information about an
 * [Assessment] to allow referencing the resource information needed to load the actual [Assessment] on demand. This
 * allows using a "placeholder" that can be vended from a different source from the actual assessment. For example,
 * an active task may be defined using a hardcoded JSON file and included in a module but requested via a
 * [TransformableAssessment] that is vended from a server.
 */
interface TransformableAssessment : Assessment, TransformableNode {
    override fun unpack(moduleInfoProvider: ModuleInfoProvider, resourceInfo: ResourceInfo, jsonCoder: Json): Assessment {
        val assessment = moduleInfoProvider.getRegisteredAssessment(this)
        return if (assessment != null) assessment else {
            val curResourceInfo = moduleInfoProvider.getRegisteredResourceInfo(this) ?: resourceInfo
            val serializer = PolymorphicSerializer(Assessment::class)
            val jsonString = moduleInfoProvider.fileLoader.loadFile(this, resourceInfo)
            val curJsonCoder = moduleInfoProvider.getRegisteredJsonDecoder(this) ?: jsonCoder
            val unpackedNode = curJsonCoder.decodeFromString(serializer, jsonString)
            unpackedNode.unpack(moduleInfoProvider.fileLoader, curResourceInfo, curJsonCoder)
        }
    }

    override fun getNavigator(): Navigator = throw IllegalStateException("Attempted to get a navigator without first unpacking the Assessment.")
}
```